keras/applications/nasnet.py

```diff
@@ -167,7 +167,7 @@ def NASNet(input_shape=None,
                                       default_size=default_size,
                                       min_size=32,
                                       data_format=K.image_data_format(),
-                                      require_flatten=include_top or weights,
+                                      require_flatten=include_top,
```

Comment:
Indeed.

Reply:
@fchollet and @gabrieldemarmiesse: It seems to me that both `nasnet` and `xception` should **not** have `require_flatten=False`. In the situation where we want to `include_top`, `_obtain_input_shape` will return `(None, None, None, 3)` rather than `(None, default_size, default_size, 3)`. As a result, the user would have to hard code the input shape, in their image preprocessing steps, for each model. Worse, if the user doesn't realize that the input shape needs to be fixed, the error that would be thrown, as a result of predicting on an image with the wrong shape, may not be very helpful.

Am I missing something here?